src/vs/platform/quickinput/browser/quickPickPin.ts

```diff
@@ -34,14 +34,16 @@ export async function showWithPinnedItems(accessor: ServicesAccessor, storageKey
 function _formatPinnedItems(storageKey: string, quickPick: IQuickPick<IQuickPickItem>, storageService: IStorageService, changedItem?: IQuickPickItem): QuickPickItem[] {
 	const formattedItems: QuickPickItem[] = [];
 	const labels = getPinnedItems(storageKey, storageService).map(item => item.label);
-	const updatedLabels = !!changedItem?.label ? updatePinnedItems(storageKey, changedItem, storageService, new Set(labels).has(changedItem.label)) : labels.filter(l => l !== 'Pinned');
+	const updatedLabels = !!changedItem?.label ? updatePinnedItems(storageKey, changedItem, storageService, new Set(labels).has(changedItem.label)) : labels;
```

Comment:
Shouldn't this `Set` check the `id`?

Reply:
hmm no because might not have an ID (like our terminal ones do not) before they're pinned